### PR TITLE
fix(auth): handle non-Transport DefaultTransport

### LINKED ISSUE #10742: auth: panic when calling httptransport.NewClient if http.DefaultTransport was overwritten
## Client

androidmanagement/v1

## Environment

go 1.19

## Code and Dependencies

```go
package main

import (
    "net/http"
    "cloud.google.com/go/auth/httptransport"
)

type rt struct {
	key   string
	value string
}
func (r *rt) RoundTrip(req *http.Request) (*http.Response, error) {
	req2 := req.Clone(req.Context())
	req2.Header.Add(r.key, r.value)
	return http.DefaultTransport.RoundTrip(req2)
}

func main() {
    // Override the global http.DefaultTransport.
    http.DefaultTransport = &rt{}

    _, err := httpttransport.NewClient(&httptransport.Options{
        APIKey: "key",
    })
    if err != nil {
      return
    }
}
```

## Expected behavior

Calling `httptransport.NewClient` with a custom implementation of `http.DefaultTransport` should not panic.

## Actual behavior

```panic: interface conversion: http.RoundTripper is *rt, not *http.Transport```

## Additional context

If a client does not specify opts.BaseRoundTripper in httptransport.NewClient, avoid a panic if the global http.DefaultTransport has been overwritten to something that is not an instance of *http.Transport.

It seems that there are several cases where this type check occurs, but this one in particular is blocking our upgrade of Google APIs.

Prior art:
- #10638 (fixed by #10669)
- #10159 (fixed by #10162)

Our use case is to create a new Google API client, but the constructor for NewClient does not allow passing a BaseRoundTripper option as seen [here](https://github.com/googleapis/google-api-go-client/blob/v0.193.0/transport/http/dial.go#L51).

CC @codyoss - proposed PR to fix is here: https://github.com/googleapis/google-cloud-go/pull/10733

### PR DESCRIPTION
**Pull request by adapap**
If a client does not specify opts.BaseRoundTripper in `httptransport.NewClient`, avoid a panic if the global http.DefaultTransport has been overwritten to something that is not an instance of *http.Transport.

Our use case is to create a new Google API client, but the constructor for NewClient does not allow passing a `BaseRoundTripper` option as seen [here](https://github.com/googleapis/google-api-go-client/blob/v0.193.0/transport/http/dial.go#L51).

Fixes #10742.

### CODE REVIEW

**Comment by codyoss** [comment]
@adapap Can you please create an issue for discussing the change, thanks.

**Comment by adapap** [comment]
@codyoss Based on our discussion in #10742, I've updated this PR to export `BaseTransport` from the `auth/internal/transport` package, and use this as the fallback instead of accepting the `http.DefaultTransport` if it was modified.

File: auth/internal/transport/transport.go [hunk 1]
```diff
@@ -81,12 +81,12 @@ func ValidateUniverseDomain(clientUniverseDomain, credentialsUniverseDomain stri
...
 // DefaultHTTPClientWithTLS constructs an HTTPClient using the provided tlsConfig, to support mTLS.
 func DefaultHTTPClientWithTLS(tlsConfig *tls.Config) *http.Client {
-	trans := baseTransport()
+	trans := BaseTransport()
 	trans.TLSClientConfig = tlsConfig
 	return &http.Client{Transport: trans}
 }
 
-func baseTransport() *http.Transport {
+func BaseTransport() *http.Transport {
```
[the remark this answers is not in the document]
Comment (adapap) [reply auth/internal/transport/transport.go:91]: Added!

**Review by codyoss — APPROVED** [review]

**Comment by codyoss** [comment]
@adapap Thank you for your contribution!

**Pull request merged**